pkg/apis/operations/types.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operations
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// Object is an operations object resource.
+type Object interface {
+	metav1.Object
+	// GetProviderType gets the type of the provider.
+	GetProviderType() string
+}
```

Comment:
Why is this interface needed?

Reply:
I had to make many decisions as to what would probably be "boilerplate" for any new API  group in Gardener. Copying this interface, as well as all the Condition stuff, constants etc. was done because I did not want to introduce cross-group references (i.e. the operations group using core stuff). Coin toss went wrong, apparently.

Reply:
OK, I see. Generally, I guess there's no reason why cross-group references should be avoided (we have this already in a lot of places and the majority of native Kubernetes types is similar). As for this interface - it's not needed at this point (you probably copied from the `extensions` group, I assume, though, this one is different compared to `core`/`settings`/`seedmanagement` because it's not served by GAPI but CRDs and some other conditions apply).
So, if you can reuse what we already have, then let's do this for now :)